src/couch_query_servers.erl

```diff
@@ -41,7 +41,7 @@ try_compile(Proc, FunctionType, FunctionName, FunctionSource) ->
     catch {compilation_error, E} ->
         Fmt = "Compilation of the ~s function in the '~s' view failed: ~s",
         Msg = io_lib:format(Fmt, [FunctionType, FunctionName, E]),
-        throw({compilation_error, Msg})
+        throw({bad_request, compilation_error, Msg})
```

Comment:
I think you should change `couch_db:validate_ddoc` instead. Because this function may be called for the valid user request while complication will fail and then CouchDB will lie again.